source/MQTTFileDownloader.c

```diff
@@ -335,30 +376,156 @@ static MQTTFileDownloaderStatus_t handleCborMessage( uint8_t * decodedData,
     return handleStatus;
 }
 
-static MQTTFileDownloaderStatus_t handleJsonMessage( uint8_t * decodedData,
-                                                     size_t * decodedDataLength,
-                                                     uint8_t * message,
-                                                     size_t messageLength )
+static bool getNumberFromString( const char * str,
+                                 size_t len,
+                                 int32_t * num )
+{
+    int32_t out = 0;
+    int32_t digit;
+    bool retVal = true;
+    const int32_t maxValue = 2147483647;
+
+    /* Biggest number which can fit in an int32_t is 2147483647 which has 10 digits. */
+    assert( len <= 10 );
+
+    for( size_t uxIndex = 0U; uxIndex < len; uxIndex++ )
+    {
+        if( IS_CHAR_DIGIT( str[ uxIndex ] ) == true )
+        {
+            digit = CHAR_TO_DIGIT( str[ uxIndex ] );
+
+            if( ( maxValue / 10 ) < out )
+            {
+                /* The out value will overflow on multiplication with 10. */
+                retVal = false;
+            }
+            else if( ( maxValue - digit ) < ( out * 10 ) )
+            {
+                /* The value ( out * 10 ) will overflow when the digit is
+                 * added to it. */
+                retVal = false;
+            }
+            else
+            {
+                out = ( out * 10 ) + digit;
+            }
+        }
+        else
+        {
+            retVal = false;
+        }
+
+        if( retVal == false )
+        {
+            break;
+        }
```

Comment:
Would probably make sense to just turn this for loop into a while loop and then do a paired conditional with this, or to just include this check in the for loop
Only pointing it out because of the "if without an else MISRA issue."

Reply:
MISRA allows an if without an else.
MISRA doesn't allow if - else-if without an else.